app/src/main/java/com/goldenraven/padawanwallet/ui/wallet/QRScanScreen.kt

```diff
@@ -97,6 +97,7 @@ internal fun QRScanScreen(navController: NavHostController) {
                                             navController.previousBackStackEntry
                                                 ?.savedStateHandle
                                                 ?.set("BTC_Address", it)
+                                            imageAnalysis.clearAnalyzer()
```

Comment:
I'm not quite sure _why_ this is solving the issue. Can you explain more what you tried and how you got to this solution?

Reply:
@thunderbiscuit Actually when we click the camera button the `QRCodeAnalyzer`'s result starts firing continuously, if there is no QR code it will go to the catch statement, and if a valid QR gets detected, it will set the address of the QR. But unfortunately even after getting a valid QR, `QRCodeAnalyzer`'s result fires extra one or two times. After getting the first valid QR `navController.popBackStack()` line gets executed and the QR Screen returns to Send Screen. After that when the `QRCodeAnalyzer`'s result fires again, `navController.popBackStack()` line gets executed again, and Send Screen returns to the Wallet screen, this process happens so fast that, To the user, it seems the QR screen directly returns to Wallet screen. what I did here is, after getting the first valid QR I cleared the `imageAnalysis` so that after getting the first valid QR it stops firing the `QRCodeAnalyzer`'s result.

Reply:
Ok cool perfect. That makes total sense, nice catch.